nncf/quantization/algorithms/bias_correction/algorithm.py

```diff
@@ -227,40 +210,63 @@ def _get_subgraph_data_for_node(self, node: NNCFNode, nncf_graph: NNCFGraph) ->
         :param nncf_graph: NNCFGraph instance for graph analysis.
         :return: A dict with the list of the nodes for the subgraph input and statistics collection.
         """
-        stats_nodes, input_nodes, output_nodes = [], [], []
+        statistic_nodes, subgraph_input_nodes, subgraph_output_nodes = [], [], []
 
-        def traverse_to_layers_with_bias(node, output):
-            if node in output:
-                return True, output
-            if self._backend_entity.is_node_with_bias(node, nncf_graph):
-                output.append(node)
-                self._collected_stat_inputs.add(node.node_name)
-                activation_input = nncf_graph.get_input_edges(node)[0].from_node
-
-                output_nodes.append(activation_input)
-                return True, output
-            return False, output
-
-        def traverse_to_input_layers(node, output):
-            if node in output + input_nodes:
-                return True, output
-            if node.node_name in self._collected_stat_inputs and node not in stats_nodes:
-                output.append(node)
-                return True, output
-            return False, output
+        def fill_statistic_nodes(node):
+            # A small hack to speed up graph traversal.
+            if node in statistic_nodes or node in visited_nodes:
+                return
+            visited_nodes.append(node)
 
+            # If we found a node with bias, we have to collect it as a statistic node,
+            # and its input for _collected_stat_inputs_map,
+            # which will be used during the collection of statistics for the next node.
+            if self._backend_entity.is_node_with_bias(node, nncf_graph) and self._backend_entity.is_quantized_weights(
```

Comment:
For now, yes, the bias correction applies only to quantized operations.
Null biases insertion, for now, gives us no value, only for BC algo speed up.

Reply:
I remembered the discussion about speed up of BC algorithm. Could you find another solution to avoid inserting null biases to the model. One of the solution can be create a map with bias operations or update the algorithm.

Reply:
It's not possible to improve this moment due to specificity of the algorithm.

Reply:
Ticket - 113700